src/backend/defines.hpp

```diff
@@ -41,3 +41,11 @@ clipFilePath(std::string path, std::string str)
     #define __AF_FILENAME__ (clipFilePath(__FILE__, "src/").c_str())
 #endif
 
+typedef enum {
+    AF_BATCH_UNSUPPORTED = -1, /* invalid inputs */
+    AF_BATCH_NONE,             /* one signal, one filter   */
+    AF_BATCH_SIGNAL,           /* many signal, one filter  */
+    AF_BATCH_KERNEL,           /* one signal, many filter  */
+    AF_BATCH_SAME,             /* signal and filter have same batch size */
+    AF_BATCH_DIFF,             /* signal and filter have different batch size */
```

Comment:
Can you change the name to `AF_BATCH_LHS` and `AF_BATCH_RHS` instead of signal and kernel ?